config/v1/types_apiserver.go

```diff
@@ -63,6 +66,39 @@ type APIServerNamedServingCert struct {
 	ServingCertificate SecretNameReference `json:"servingCertificate"`
 }
 
+type APIServerEncryption struct {
+	// type defines what encryption type should be used to encrypt resources at the datastore layer.
+	// When this field is unset (i.e. when it is set to the empty string), identity is implied.
+	// The behavior of unset can and will change over time.  That is, encryption will be turned
+	// on by default in a future release.  Even after encryption is enabled by default, the meaning
+	// of unset may change to a different encryption type based on changes in best practices.
+	//
+	// When encryption is enabled, all sensitive resources shipped with the platform are encrypted.
+	// This list of sensitive resources can and will change over time.  The current authoritative list is:
+	//
+	//   1. secrets
+	//   2. config maps
+	//   3. routes
+	//   4. oauth access tokens
+	//   5. oauth authorize tokens
+	//
+	// +unionDiscriminator
+	// +optional
+	Type EncryptionType `json:"type,omitempty"`
+}
+
+type EncryptionType string
+
+const (
+	// identity refers to a type where no encryption is performed at the datastore layer.
+	// Resources are written as-is without encryption.
+	EncryptionTypeIdentity EncryptionType = "identity"
```

Comment:
isn't `"none"` better? Identity sounds pseudo scientific.

Reply:
or `"disabled"`

Reply:
isn't setting empty string better for "no" encryption? are we going to "default" to "identity" here or we default to nothing (the omitempty above suggests that). I guess user can just remove the encryption type when they don't want any encryption?

Reply:
we want to change the default later, at least keep us that possibility. Therefore we distinguish here between "explicit no encryption" and the empty string.

Reply:
Yeah unset means "the platform does what it feels like is the best thing."  You get to override that if you want to.

Reply:
The name of the type is based on how it is referred to upstream.  I do not think we should diverge from that in this case.

https://kubernetes.io/docs/tasks/administer-cluster/encrypt-data/#providers

Reply:
Am fine with identity.